test/NFTOpt.tests.ts

```diff
@@ -286,9 +287,194 @@ describe("NFTOpt Tests", function () {
     });
 
     describe("cancelOption", function () {
+
         it("should test that method can be called", async function () {
             expect(NFTOptCTR.connect(buyer).cancelOption(0)).to.not.throw;
         });
+
+        it("should test that the option exists", async function () {
+
+            expect(
+                NFTOptCTR.connect(buyer).cancelOption(0)
+                ).to.be.revertedWith("The Option does not exist");
+        });
+
+        it("should revert when the option does not have a start date i.e. published: REQUEST state", async function () {
+
+            await NFTOptCTR.connect(buyer).publishOptionRequest
```

Comment:
there is a method wrapper for this called `publishOptionRequestDummy()` use that instead, it's not needed here to test an American option specifically